[CS2113-T14-4] DietaryBook

tikimonarch opened the pull request:
DietBook helps the NUS student community specifically to manage their daily dietary intakes with tracking, calculation and meaningful advise/suggestions. It is optimized for CLI users so that frequent tasks can be done faster by typing in commands.

tikimonarch committed:
Update ExitCommand.java

HengFuYuen committed:
Merge pull request #104 from tikimonarch/MainBranch

Major feature updated

HengFuYuen committed:
Add case where gender is male

HengFuYuen committed:
Merge pull request #105 from HengFuYuen/IncludeMale

Add case where gender is male

tikimonarch committed:
Bug Fixes

HengFuYuen committed:
Merge pull request #106 from tikimonarch/MainBranch

Bug Fixes

HengFuYuen committed:
Update help command message

HengFuYuen committed:
Merge pull request #107 from HengFuYuen/Ui-EditDatabase

Update help command message

0xEljh committed:
add addFoodAtDateTime

Method overload that accepts food object directly. For save-loading and adding from database.

tikimonarch committed:
Merge pull request #108 from AY2021S1-CS2113-T14-4/clean-up-branch

overload addFoodAtDateTime

snowbanana12345 committed:
Put Examples of Usage for adding controling food items into the User Guide

snowbanana12345 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp

yeapcl commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,155 @@
+![Alt text](save_load_feature/FoodSaveLoadManager_save.png)
+##### FoodSaveLoadManager#load()
+![Alt text](save_load_feature/FoodSaveLoadManager_load.png)
+similiar diaghrams for PersonSaveLoadManager
```
same as above, should classes be placed under the activation bar? It may not be non-standard.
![image](https://user-images.githubusercontent.com/69618398/97671467-48f6a280-1ac3-11eb-8292-2474f5c24423.png)

yeapcl commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,155 @@
 
+## Design 
+
+### UI component
+![Ui component](diagrams/Ui component.png)
+
```
Should the arrow pointing to enumeration be dotted?

![image](https://user-images.githubusercontent.com/69618398/97530119-60169100-19ec-11eb-9746-a3b9ee83340b.png)

HengFuYuen replied:
Thanks for reviewing! Yes, Ui has dependencies and not associations with the enum classes which is why I used the dashed lines. Perhaps my following explanations were not clear enough. Thanks for the feedback!

snowbanana12345 committed:
Add examples of usage to UserGuide section

snowbanana12345 committed:
minor fixes

tikimonarch committed:
Merge pull request #109 from snowbanana12345/Zhong-Ming-UserGuide-Branch

Zhong ming user guide branch

tikimonarch committed:
Update UserGuide.md

tikimonarch committed:
Merge branch 'master' into MainBranch

HengFuYuen committed:
Merge pull request #110 from tikimonarch/MainBranch

Update UserGuide.md

tikimonarch committed:
Add command with date time

HengFuYuen committed:
Merge pull request #111 from tikimonarch/MainBranch

Add command with date time

tikimonarch committed:
Future date exception

HengFuYuen committed:
Update _config.yml

HengFuYuen committed:
Merge pull request #112 from tikimonarch/MainBranch

Future date exception

HengFuYuen committed:
Fix major formatting issues

HengFuYuen committed:
Merge pull request #113 from HengFuYuen/UpdateGuide

Fix major formatting issues

HengFuYuen committed:
Fix formatting

HengFuYuen committed:
Merge pull request #114 from HengFuYuen/Guide

Fix formatting

zsk612 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,155 @@
+The UI has a dependency with two enumeration class, [`ActivityLevel`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master/src/main/java/seedu/dietbook/person/ActivityLevel.java) and [`Gender`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master/src/main/java/seedu/dietbook/person/Gender.java) as descriptions of each [`ActivityLevel`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master ...
+
+## Implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```
Perhaps you should delete this line for the final version?

HengFuYuen replied:
Noted! Thank you!

zsk612 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,155 @@
+
+`name` command
+
+![Name sequence diagram](diagrams/Name sequence diagram.png)
```
Perhaps the constructor block is wrong.

HengFuYuen replied:
Noted! Thanks for spotting the error!

zsk612 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,155 @@
+
+`name` command
+
+![Name sequence diagram](diagrams/Name sequence diagram.png)
```
Perhaps you should align the dotted line to the middle of the execute() block.

HengFuYuen replied:
Noted! Thanks for spotting the error!

tikimonarch committed:
Update UserGuide.md

zsk612 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,155 @@
+* PersonSaveLoadManager#load() loads the file 
+* Setters and Getters for all the personal data in this current version
+
+#### UML diaghram
+##### FoodSaveLoadManager#save()
+![Alt text](save_load_feature/FoodSaveLoadManager_save.png)
```
Perhaps you should end the dotted line at the cross

yeapcl replied:
Are classes supposed to be below activation bars? It may be non-standard
![image](https://user-images.githubusercontent.com/69618398/97671395-1d73b800-1ac3-11eb-9cc6-65eee175ff0b.png)

yeapcl commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,155 @@
+* FoodSaveLoadManager#save() saves the list of food objects to the specified file name
+* FoodSaveLoadManager#load() loads the file and returns the list of food objects stored inside it
+
+#### PersonSaveLoadManager class
+Built on top of Saver and Loader class to implement save/load functionality for user information
+Same as FoodSaveLoadManager, it has its own folder to work with, the user only has to specify the file name
+Unlike the FoodSaveLoadManager, it stores the data inside itself and can be updated.
```
perhaps a class and sequence diagram would help convey the idea better idea.

yeapcl reviewed (commented):
Overall quite good, but with a few minor suggestions/clarifications needed

HengFuYuen committed:
Merge pull request #115 from tikimonarch/MainBranch

Update UserGuide.md

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp

* 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp: (82 commits)
  Update UserGuide.md
  Fix formatting
  Fix major formatting issues
  Update _config.yml
  Future date exception
  Add command with date time
  Update UserGuide.md
  minor fixes
  Add examples of usage to UserGuide section
  Put Examples of Usage for adding controling food items into the User Guide
  add addFoodAtDateTime
  Update help command message
  Bug Fixes
  Add case where gender is male
  Update ExitCommand.java
  Database Printing
  Feature Updates
  add toString version of datetime support methods
  Add dataSuccessfullySavedMessage method
  Delete Food.java in Test
  ...

Commit:
Added a CalculatorData class to take in a FoodList and return information needed to calculate to Calculator class.

Added the initialization of CalculatorData class in manager.
Deleted setCalculator method in manager.

Changed the inputs of methods related with calculator in CalculateCommand class.

Commit:
added update method in Calculator class.

Commit:
added the setCalculator method

Commit:
Added the initialization of data.

0xEljh committed:
add dated string representation

- The datetimes of the entries in FoodList can now be displayed to the user.
- Filtering the list by datetime is now start and end inclusive.

0xEljh committed:
FoodListTest improvements

0xEljh committed:
Remove unused imports

Cleanup of master branch: Removed imports that are unused or duplicated — likely due to merge conflicts.

HengFuYuen committed:
Rename ActivityLevel to FitnessLevel

HengFuYuen committed:
Add logging for Person class

snowbanana12345 committed:
edit data base

snowbanana12345 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp into Zhong-ming-Branch-DataBase

snowbanana12345 committed:
update data base

snowbanana12345 committed:
rename architecture to design
architecture refers to high level architecture of the software

snowbanana12345 committed:
change activity level to fitness level in person class

HengFuYuen committed:
Merge pull request #164 from HengFuYuen/PersonAndFitness

Person and fitness

HengFuYuen committed:
Merge branch 'master' into clean-up-branch

HengFuYuen committed:
Merge pull request #163 from AY2021S1-CS2113-T14-4/clean-up-branch

[Issue: 137] Add dated string representation of FoodList and cleanups

tikimonarch committed:
Exceptions

-Catching all exception
-Fixed for specific exceptions

tikimonarch committed:
Bug fixes

-fixed calculate rubbish acceptance
- fixed accepting empty name/empty param checker

tikimonarch committed:
Add fixed

-all option swappable but time must be last

HengFuYuen committed:
Refactor Ui and add logging

HengFuYuen committed:
Change global variable to local

HengFuYuen committed:
Add new line at the end of the file

HengFuYuen committed:
Update JavaDoc comments

HengFuYuen committed:
Update I/O testing

tikimonarch committed:
Update InputChecker.java

tikimonarch committed:
Improved exception catching

tikimonarch committed:
Improved option detection

tikimonarch committed:
Updated error message

HengFuYuen committed:
Merge pull request #169 from HengFuYuen/PersonLogging

Change global variable to local

tikimonarch committed:
Improved command exception

- Specific exception messages

tikimonarch committed:
Fixed End Time

tikimonarch committed:
Merge pull request #166 from tikimonarch/MainBranch

Bug Fixes

HengFuYuen committed:
Update README.md

HengFuYuen committed:
Delete DukeTest.java

HengFuYuen committed:
Update JavaDoc comments

HengFuYuen committed:
Add MainLogger.java

HengFuYuen committed:
Add logging for Ui.java

HengFuYuen committed:
Add logging for UiHelper.java

HengFuYuen committed:
Use MainLogger for UiInput.java

HengFuYuen committed:
Add logging for UiMessage.java

HengFuYuen committed:
Make UiOutput package private

HengFuYuen committed:
Make method package private

HengFuYuen committed:
Update JUnit tests for UiHelper

HengFuYuen committed:
Add JUnit tests for UiInput

HengFuYuen committed:
Merge branch 'master' into UiChanges

HengFuYuen committed:
Merge pull request #170 from HengFuYuen/UiChanges

Refactor Ui and add logging

HengFuYuen committed:
Use MainLogger for logging

snowbanana12345 committed:
add resource to test folder

snowbanana12345 committed:
fix check style

HengFuYuen committed:
Add attribution

HengFuYuen committed:
Add assertions for MainLogger

HengFuYuen committed:
Add JavaDoc comments for MainLogger

HengFuYuen committed:
Update AboutUs.md

HengFuYuen committed:
Add PPP

HengFuYuen committed:
Update introduction

HengFuYuen committed:
Update quick start

HengFuYuen committed:
Add notes and warnings to features section

HengFuYuen committed:
Update name command feature

HengFuYuen committed:
Update name command feature

HengFuYuen committed:
Update info command feature

HengFuYuen committed:
Update editinfo command feature

HengFuYuen committed:
Update minor formatting issues

HengFuYuen committed:
Update command summary

HengFuYuen committed:
Update image

HengFuYuen committed:
Update code quality

Commit:
Changed get... methods in CalculatorData class to getPortioned... methods.

snowbanana12345 committed:
implement and test search food by index function in data base

tikimonarch committed:
Exception Bug Fixes

- calculate error message fixed
- list exception message fixed
- single word command fixed
    ->help
    ->recommend
    ->exit
    ->userinfo
    ->data
    ->clear
-fixed time detection

snowbanana12345 committed:
passed check style

snowbanana12345 committed:
fix check style

snowbanana12345 committed:
no message

snowbanana12345 committed:
fix test check style

tikimonarch committed:
Merge branch 'master' into MainBranch

tikimonarch committed:
Merge pull request #179 from tikimonarch/MainBranch

Exception Bug Fixes

HengFuYuen committed:
Change preposition

snowbanana12345 committed:
start implementing FoodPortionDateSaveLoadManager

snowbanana12345 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp into Zhong-ming-saveload-branch

HengFuYuen committed:
Update grammar and divider length

yuqiaoluolong committed:
Merge branch 'master' into branch-3-changes

yuqiaoluolong committed:
Merge pull request #162 from yuqiaoluolong/branch-3-changes

changes in calculator

Commit:
Changed the the ActivityLevel to FitnessLevel

Commit:
Changed the input attributes in the calculate... methods in CalculateCommand class.

HengFuYuen committed:
Change help command output

HengFuYuen committed:
Merge branch 'master' into UiUpdates

HengFuYuen committed:
Merge pull request #180 from yuqiaoluolong/branch-4-changes

Changed the the ActivityLevel to FitnessLevel

HengFuYuen committed:
Merge branch 'master' into UiUpdates

HengFuYuen committed:
Update I/O testing

0xEljh committed:
add support for missing food details

- FoodList can now accept entries with missing values.
- Missing values are recalulated upon the use of a getFood() method call.
- add a string utility class for fstring-like support.

0xEljh committed:
move StringFormatter to utils package

StringFormatter is a utility class that is not limited to supporting classes in list.

0xEljh committed:
relax FoodListTest

Remove reliance on time elapsed after datetime.now()

0xEljh committed:
Remove unused dependencies

HengFuYuen committed:
Update notes and warnings under Feature section

HengFuYuen committed:
Elaborate on valid inputs for user name

HengFuYuen committed:
Combine notes and warning section

HengFuYuen committed:
Merge pull request #173 from HengFuYuen/UpdatePersonLogging

Use MainLogger for logging

HengFuYuen committed:
Merge branch 'master' into AboutUs-PPP

snowbanana12345 committed:
implement 2nd layer of setters and getters

snowbanana12345 committed:
add basic functionality test for FoodPortionSaveLoadManager

snowbanana12345 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp into Zhong-ming-saveload-branch

snowbanana12345 committed:
implement and test save loading food list

snowbanana12345 committed:
pass check style

snowbanana12345 committed:
copy resources over to test

HengFuYuen committed:
Update help command output

tikimonarch committed:
Improved date time check

tikimonarch committed:
Merge branch 'master' into MainBranch

tikimonarch committed:
Bug fixes for garbage input

- delete 3 garbage
- list garbage
- calculate carb garbage

snowbanana12345 committed:
Merge pull request #185 from snowbanana12345/Zhong-ming-Branch-DataBase

Zhong ming branch data base

HengFuYuen committed:
Merge pull request #171 from HengFuYuen/README

Update README.md

HengFuYuen committed:
Merge pull request #178 from HengFuYuen/UpdateUserGuide

Update user guide

HengFuYuen committed:
Fix icon

HengFuYuen committed:
Improve formatting

tikimonarch committed:
Multi forward slash detection

Commit:
Added the minimum and maximum caps for recommended calorie intake.

HengFuYuen committed:
Merge pull request #190 from HengFuYuen/UserGuideUpdate

User Guide

HengFuYuen committed:
Merge pull request #188 from HengFuYuen/UpdateReadMe

Fix icon

0xEljh committed:
add sort

FoodList can be sorted by datetime of its entries

0xEljh committed:
Merge pull request #183 from AY2021S1-CS2113-T14-4/FoodList-Optional-Branch

[Issue: 182] Support for optional fields in FoodList

Commit:
Created a testing method corresponding to the added caps.

Commit:
refactored the calculateRecommendation to improve the code quality.

HengFuYuen committed:
Merge pull request #192 from AY2021S1-CS2113-T14-4/FoodList-Optional-Branch

[Issue: 88] Add sort

Commit:
Changed the weight used in calculateRecommendation method from original ones to current ones.

Commit:
change the activity stuff to fitness...

Commit:
activity to fitness

Commit:
Changed some numbers accordingly.

yuqiaoluolong committed:
Merge pull request #191 from yuqiaoluolong/branch-5-cap

Added the minimum and maximum caps for recommended calorie intake.

HengFuYuen committed:
Update the Ui component description

HengFuYuen committed:
Update Ui component diagram

HengFuYuen committed:
Merge branch 'master' into DeveloperGuide

HengFuYuen committed:
Update enter user information feature description

tikimonarch committed:
Merge branch 'master' into MainBranch

HengFuYuen committed:
Update enter user information feature diagrams

HengFuYuen committed:
Update user stories related to Person class

HengFuYuen committed:
Update non-functional requirements

HengFuYuen committed:
Update glossary

HengFuYuen committed:
Update target user profile and value preposition

HengFuYuen committed:
Update feature description

HengFuYuen committed:
Update edit user info feature

0xEljh committed:
update DevGuide with Model

- small tweak to DatedFoodEntry for consistency
- Added Model section to Design section of dev guide

HengFuYuen committed:
Rename files to remove spacing

HengFuYuen committed:
Update edit user information feature

HengFuYuen committed:
Update view user information feature

HengFuYuen committed:
Update instructions for manual testing

HengFuYuen committed:
Update _config.yml

HengFuYuen committed:
Update HelpCommand display for user guide

HengFuYuen committed:
Merge pull request #194 from HengFuYuen/DeveloperGuide

Developer guide

HengFuYuen committed:
Merge branch 'master' into AboutUs-PPP

HengFuYuen committed:
Fix code snippet formatting with image

HengFuYuen committed:
Improve code quality

tikimonarch committed:
Data adding

tikimonarch committed:
Merge branch 'master' into MainBranch

tikimonarch committed:
Refactored Calculate command

HengFuYuen committed:
Update multiplicities in the diagram

HengFuYuen committed:
Add hengfuyuen.md

HengFuYuen committed:
Merge pull request #176 from HengFuYuen/AboutUs-PPP

HengFuYuen-AboutUs.md,PPP and Developer Guide

HengFuYuen committed:
Merge branch 'master' into UiUpdates

Commit:
Some changes about the calculator parts in UG.

HengFuYuen committed:
Update AboutUs.md

HengFuYuen committed:
Merge pull request #198 from HengFuYuen/AboutUs

Update AboutUs.md

tikimonarch committed:
data base adding fixed

Commit:
changed the format about us back.

snowbanana12345 committed:
Merge branch 'master' into Zhong-ming-saveload-branch

snowbanana12345 committed:
Merge pull request #184 from snowbanana12345/Zhong-ming-saveload-branch

Zhong ming saveload branch

HengFuYuen committed:
Merge pull request #197 from yuqiaoluolong/branch-6-UG

Some changes about the calculator parts in UG.

HengFuYuen committed:
Update PRs

0xEljh committed:
add details to model section

HengFuYuen committed:
Fix formatting errors

HengFuYuen committed:
Merge branch 'master' into PPPAndUG

# Conflicts:
#	docs/UserGuide.md

HengFuYuen committed:
Update PPP

HengFuYuen committed:
Merge branch 'master' into UiUpdates

HengFuYuen committed:
Remove TOC

tikimonarch committed:
Merge branch 'master' into MainBranch

snowbanana12345 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp into Zhong-Ming-UserGuide-Branch

snowbanana12345 committed:
update dev guide documentation.
update UML diaghrams

0xEljh committed:
add missing fields feature to implementation

snowbanana12345 committed:
create dev guide section on database

Commit:
added person info in AboutUs

HengFuYuen committed:
Merge pull request #200 from yuqiaoluolong/branch-7-AboutUs

added person info in AboutUs

HengFuYuen committed:
Merge pull request #199 from HengFuYuen/PPPAndUG

Ppp and ug

snowbanana12345 committed:
data base uml diaghrams

snowbanana12345 committed:
data base segment done

tikimonarch committed:
Implemented Save/Load

tikimonarch committed:
Merge branch 'master' into MainBranch

snowbanana12345 committed:
start writing PPP

snowbanana12345 committed:
update PPP

snowbanana12345 committed:
Merge pull request #201 from snowbanana12345/Zhong-Ming-UserGuide-Branch

Zhong ming user guide branch

Commit:
Condense the calculator part in UG

snowbanana12345 committed:
update PPP

0xEljh committed:
add uml sequence diagrams on OptionalFood feature

tikimonarch committed:
Merge pull request #186 from tikimonarch/MainBranch

Improved date time check, more improved exception catching

snowbanana12345 committed:
Merge pull request #203 from snowbanana12345/Zhong-Ming-UserGuide-Branch

update PPP

HengFuYuen committed:
Fix bugs

HengFuYuen committed:
Merge branch 'master' into UiUpdates

Commit:
recalculation

yuqiaoluolong committed:
Merge pull request #202 from yuqiaoluolong/branch-8-condense_UG

Condense the calculator part in UG

Commit:
PPP

0xEljh committed:
add links to source files

links added as comments in markdown

0xEljh committed:
Merge branch 'master' into docs

snowbanana12345 committed:
update data base section of user guide
update PPP

snowbanana12345 committed:
update user guide FAQ section on folder

yuqiaoluolong committed:
Merge pull request #206 from yuqiaoluolong/branch-9-PPP

PPP

snowbanana12345 committed:
Merge pull request #207 from snowbanana12345/Zhong-Ming-UserGuide-Branch

Zhong ming user guide branch

Commit:
some edits on ppp

HengFuYuen committed:
Remove unused methods

HengFuYuen committed:
Merge pull request #181 from HengFuYuen/UiUpdates

Ui updates

0xEljh committed:
Merge pull request #204 from AY2021S1-CS2113-T14-4/docs

Update Dev Guide

HengFuYuen committed:
Merge pull request #205 from HengFuYuen/PPP

Fix bugs

Commit:
some edits on UG about recalculation

yuqiaoluolong committed:
Merge pull request #208 from yuqiaoluolong/master

some edits on ppp

0xEljh committed:
create elijahng project portfolio

0xEljh committed:
fix markdown comments

0xEljh committed:
Revert "fix markdown comments"

This reverts commit 6920dd945f17f3ae694479d3f28dd8661c6f5ce8.

0xEljh committed:
fix markdown comments

HengFuYuen committed:
Merge pull request #210 from AY2021S1-CS2113-T14-4/docs

Comments fix

tikimonarch committed:
PPP + userguide update

0xEljh committed:
update PPP

0xEljh committed:
Merge pull request #209 from AY2021S1-CS2113-T14-4/Elijah-PPP

add project portfolio

HengFuYuen committed:
Merge pull request #211 from tikimonarch/MainBranch

PPP + userguide update

0xEljh committed:
Update AboutUs.md

tikimonarch committed:
Update AboutUs.md
